src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { FC, useState, useCallback, useMemo } from 'react';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+import { peopleFromServer } from '../../data/people';
+import { Person } from '../../types/Person';
+import { filterPeople } from '../../utils/filterPeople';
+import { DropdownMenu } from '../DropdownMenu/DropdownMenu';
+
+type Props = {
+  selectedPerson: Person | null;
+  onSelect: React.Dispatch<React.SetStateAction<Person | null>>;
+  delay?: number;
+};
+
+export const Dropdown: FC<Props> = ({
+  selectedPerson,
+  onSelect,
+  delay = 300,
+}) => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isActive, setIsActive] = useState(false);
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
+
+  const filteredPeople = useMemo(
+    () => filterPeople(peopleFromServer, { appliedQuery }),
+    [appliedQuery],
+  );
```

Comment:
change syntax and use it in the future(more readable)
```suggestion
  const filteredPeople = useMemo(() => (
    filterPeople(peopleFromServer, { appliedQuery })
  ), [appliedQuery]);
```

Reply:
I'll remember it, thanks, but here prettier changes it back